runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/External.java

```diff
@@ -222,6 +226,20 @@ public OutputT expand(InputT input) {
               });
       externalPCollectionIdMap = externalPCollectionIdMapBuilder.build();
 
+      Map<Coder, String> externalCoderIdMapBuilder = new HashMap<>();
+      expandedComponents
+          .getPcollectionsMap()
+          .forEach(
+              (pcolId, pCol) -> {
+                try {
+                  Coder coder = rehydratedComponents.getCoder(pCol.getCoderId());
+                  externalCoderIdMapBuilder.putIfAbsent(coder, pCol.getCoderId());
+                } catch (IOException e) {
+                  throw new RuntimeException("cannot rehydrate Coder.");
```

Comment:
Won't it often be the case that we can't rehydrate the coder, because it's specific to another SDK?